app/helpers/claims_helper.rb

```diff
@@ -5,49 +5,55 @@ def eligibility_answers(current_claim)
 
   def identity_answers(claim)
     [].tap do |a|
-      a << [translate("questions.name"), claim.full_name, "personal-details"] if show_name(claim)
-
-      a << [translate("questions.address.generic.title"), claim.address, "address"] unless claim.address_from_govuk_verify?
-
-      a << [translate("questions.date_of_birth"), date_of_birth_string(claim), "personal-details"] if show_dob(claim)
-
-      a << [translate("questions.payroll_gender"), t("answers.payroll_gender.#{claim.payroll_gender}"), "gender"] unless claim.payroll_gender_verified?
-
-      a << [translate("questions.teacher_reference_number"), claim.teacher_reference_number, "teacher-reference-number"] if show_trn(claim)
-
-      a << [translate("questions.national_insurance_number"), claim.national_insurance_number, "personal-details"] if show_nino(claim)
-
-      a << [translate("questions.email_address"), claim.email_address, claim.email_address_check? ? "select-email" : "email-address"]
-
-      a << [translate("questions.provide_mobile_number"), (claim.provide_mobile_number ? "Yes" : "No"), "provide-mobile-number"] if claim.has_ecp_or_lupp_policy?
-      a << [translate("questions.mobile_number"), claim.mobile_number, "mobile-number"] if claim.has_ecp_or_lupp_policy? && claim.provide_mobile_number?
+      a << [t("questions.name"), claim.full_name, "personal-details"] if show_name(claim)
+      a << [t("questions.address.generic.title"), claim.address, "address"] unless claim.address_from_govuk_verify?
+      a << [t("questions.date_of_birth"), date_of_birth_string(claim), "personal-details"] if show_dob(claim)
+      a << [t("questions.payroll_gender"), t("answers.payroll_gender.#{claim.payroll_gender}"), "gender"] unless claim.payroll_gender_verified?
+      a << [t("questions.teacher_reference_number"), claim.teacher_reference_number, "teacher-reference-number"] if show_trn(claim)
+      a << [t("questions.national_insurance_number"), claim.national_insurance_number, "personal-details"] if show_nino(claim)
+
+      a << if claim.logged_in_with_tid? && claim.teacher_id_user_info["email"].present?
+        [t("questions.select_email.heading"), claim.email_address, "select-email"]
+      else
+        [t("questions.email_address"), claim.email_address, "email-address"]
+      end
+
+      if claim.logged_in_with_tid? && claim.teacher_id_user_info["phone_number"].present?
```

Comment:
I think the `email_address_check` and `mobile_check` columns might be useful here to determine if the user provided their own details or the ones from teacher ID.

Reply:
@slorek I think you are right, but when I looked at the slug sequence [here](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/9c65314e7c4ec07f27fe9d32f01849a797916a21/app/models/journeys/additional_payments_for_teaching/slug_sequence.rb#L113-L127), it somehow seems that it's driven by `teacher_id_user_info["phone_number"]` and `teacher_id_user_info["email_address"]` (as well), so it almost seemed useless to check those `_check` flags, which would be set only if the `select-email` and `select-mobile` slugs can be rendered – which is when the user is on TID!

I'll have another look, it generally feels quite messy what's happening there and I found it easier to reason in terms of whether the mobile and email are present or not in TID.

Reply:
@slorek I had another go at this; I spent some time refactoring and adding comments, and I ended up relying on those `_check` flags as you suggested. Hopefully it's easier to understand what is the changes are trying to achieve and the various scenarios: [e7edf90](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/2634/commits/e7edf9062edfdcb791da558d1f334a975d87f667)